server/fleet/app.go

```diff
@@ -435,3 +435,8 @@ type KafkaRESTConfig struct {
 	ResultTopic string `json:"result_topic"`
 	ProxyHost   string `json:"proxyhost"`
 }
+
+// DeviceAPIFeatures specifies a list of features supported
+// by the current API version. Each field in the struct is
+// meant to be a boolean value.
```

Comment:
Also another rule: fields are never removed, only added.

Reply:
True. And it kind of makes sense, removing a feature from the struct (e.g. when deprecating a feature) will disable such feature in old installations.